src/esockd_udp.erl

```diff
@@ -308,6 +319,11 @@ handle_info(Info, State) ->
     ?ERROR_MSG("Unexpected info: ~p", [Info]),
     {noreply, State}.
 
+reinit(Sock, State) ->
```

Comment:
Should we reset all the channel pids? Channels that were started with the old sock should not be available once the sock is closed.

Reply:
in emqx_gateway_conn, the socket seems to be discarded
https://github.com/emqx/emqx/blob/34be2ea9a064ea81ec0d7024cefd71e8c4b4aec5/apps/emqx_gateway/src/bhvrs/emqx_gateway_conn.erl#L118-L125

Reply:
actually, it's used in that module, but seems only to receive the close events.